src/helm/benchmark/scenarios/melt_scenarios.py

```diff
@@ -343,3 +351,947 @@ def __init__(self, **kwargs):
+
+            if sample_idx < self.num_train_instances:
+                split = TRAIN_SPLIT
+            elif sample_idx < self.num_train_instances + self.num_val_instances:
+                split = VALID_SPLIT
+            else:
+                split = TEST_SPLIT
+
+            instance = Instance(
+                input=Input(text=question),
+                references=[Reference(Output(text=str(target_fact)), tags=[CORRECT_TAG])],
+                split=split,
+            )
+            instances.append(instance)
+
+        return instances
+
+
+ANIMALS = [
+    "con ngựa vằn",
+    "con rắn hổ mang",
+    "con cò",
+    "con chim cánh cụt",
+    "con cá mập",
+    "con sư tử",
+    "con trâu",
+    "con cá voi",
+    "con hải cẩu",
+    "con đại bàng",
+    "con ngựa",
+    "con chuột",
+]
+FRUITS = [
+    "quả táo",
+    "quả đào",
+    "quả dưa hấu",
+    "quả chuối",
+    "quả nho",
+    "quả kiwi",
+    "quả lê",
+    "quả dâu tây",
+    "quả việt quất",
+    "quả mâm xôi",
+]
+RULE_SYMBOLS = ["X", "Y", "Z"]
+MATH_SYMBOLS = ["+", "-", "*", "="]
+
+
+def subst(pattern: List[str], rule_symbol: str, substitute_str: str) -> List[str]:
+    """
+    We substitute one rule symbols in a pattern according by a substitution str.
+
+    example:
+    pattern = "A+B=B+A"
+    rule_symbol = "A"
+    substitute_str = "quả táo"
+    return: "quả táo+B=B+quả táo"
+
+    :param pattern: A Pattern representing the rule.
+    :param rule_symbol: One rule symbol.
+    :param substitute_str: The substitution string.
+    :return: The result of substitution.
+    """
+    assert rule_symbol in pattern
+    # check which index is the same as rule_symbol
+    indices = [i for i, x in enumerate(pattern) if x == rule_symbol]
+
+    # form a new string with the symbol replaced
+    new_string = pattern[: indices[0]] + [substitute_str]
+    for i, j in zip(indices[:-1], indices[1:]):
+        new_string += pattern[i + 1 : j]
+        new_string += [substitute_str]
+    new_string += pattern[indices[-1] + 1 :]
+
+    return new_string
+
+
+def pattern_subst(pattern: List[str], rule_symbols: List[str], substitute_dict: Dict[str, str]) -> List[str]:
+    """
+    We substitute the rule symbols in a pattern according to a substitution dictionary.
+
+    example:
+    pattern = "A+B=B+A"
+    rule_symbols = ["A", "B"]
+    substitute_dict = {"A":"quả táo", "B":"quả đào"}
+    return: "quả táo+quả đào=quả đào+quả táo"
+
+    :param pattern: A Pattern representing the rule.
+    :param rule_symbols: The set of rule symbols.
+    :param substitute_dict: The substitution dictionary.
+    :return: The result of substitution.
+    """
+
+    out = pattern
+    # we iteratively replace each rule symbol with its subsitution string
+    for symbol in rule_symbols:
+        out = subst(out, symbol, substitute_dict[symbol])
+    return out
+
+
+class MELTSyntheticReasoningScenario(Scenario):
```

Comment:
Please move this class and associated functions to its own file `melt_synthetic_reasoning_scenario.py`